ibm/mas_devops/roles/kubeturbo/tasks/main.yml

```diff
@@ -0,0 +1,115 @@
+
+- name: Set kubeturbo_namespace_fsgroup
+  vars:
+    regex: '.+?(?=/)'
+  set_fact:
+    kubeturbo_namespace_fsgroup: "{{ kubeturbo_namespace_output.resources[0].metadata.annotations['openshift.io/sa.scc.uid-range'] | regex_search(regex) }}"
+
+# 2. Deploy Kubeturbo subscription and Operator Group
+# -----------------------------------------------------------------------------
+- name: "Create Kubeturbo operator group"
+  kubernetes.core.k8s:
+    definition: "{{ lookup('template', 'templates/kubeturbo-operator-group.yaml') }}"
+    wait: yes
+    wait_timeout: 60 #subsequent tasks will fail if the CRD isn't fully created
+
+- name: "Create Kubeturbo subscription"
+  kubernetes.core.k8s:
+    definition: "{{ lookup('template', 'templates/kubeturbo-subscription.yaml') }}"
+    wait: yes
+    wait_timeout: 300
+    wait_condition:
+      type: 'CatalogSourcesUnhealthy'
+      status: "False"
+
+# 2. Deploy Kubeturbo CRD
+# -----------------------------------------------------------------------------
+
+- name: "Wait until the Kubeturbo CRD is available"
+  kubernetes.core.k8s_info:
+    api_version: apiextensions.k8s.io/v1
+    name: "kubeturbos.charts.helm.k8s.io"
+    kind: CustomResourceDefinition
+    wait: yes
+    wait_sleep: 10
+    wait_timeout: 300 # 5 mins until we give up waiting for the CRD to get into the expected state
+    wait_condition:
+      type: NamesAccepted
+      status: "True"
+  register: kubeturbo_crd_info
+  retries: 120 # ~approx 10 minutes before we give up waiting for the CRD to be created
+  delay: 5 # seconds
+  until:
+    - kubeturbo_crd_info.resources is defined
+    - kubeturbo_crd_info.resources | length > 0
+
+# 12. Wait until Kubeturbo Operator is ready
+# -----------------------------------------------------------------------------
+- name: "Wait for {{ kubeturbo_sub_name }} to be ready (60s delay)"
+  kubernetes.core.k8s_info:
+    api_version: apps/v1
+    name: "kubeturbo-operator"
+    namespace: "{{ kubeturbo_namespace }}"
+    kind: Deployment
+  register: kubeturbo_operator_deployment
+  until:
+    - kubeturbo_operator_deployment.resources is defined
+    - kubeturbo_operator_deployment.resources | length > 0
+    - kubeturbo_operator_deployment.resources[0].status is defined
+    - kubeturbo_operator_deployment.resources[0].status.replicas is defined
+    - kubeturbo_operator_deployment.resources[0].status.readyReplicas is defined
+    - kubeturbo_operator_deployment.resources[0].status.readyReplicas == kubeturbo_operator_deployment.resources[0].status.replicas
+  retries: 30 # Approximately 1/2 hour before we give up
+  delay: 60 # 1 minute
+
+# 13. Create Kubeturbo Custom Resource
+# -----------------------------------------------------------------------------
+- name: "Create Kubeturbo Custom Resource"
+  kubernetes.core.k8s:
+    definition: "{{ lookup('template', 'templates/kubeturbo-cr.yaml') }}"
+
+# 14. Wait Kubeturbo Custom Resourceto be complete. This needs fixing.
```

Comment:
> This needs fixing.

If this needs fixing, please fix whatever needs fixing before merging this.

Reply:
Update completed.